### PR TITLE
DNN: support GreaterOrEqual and LessOrEqual op in ONNX

### PR DESCRIPTION
**Pull request by zihaomu**
Related test data: [PR at opencv_extra](https://github.com/opencv/opencv_extra/pull/1010).

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/test/test_onnx_importer.cpp [hunk 1]
```diff
@@ -682,6 +682,16 @@ TEST_P(Test_ONNX_layers, Compare_LT)
     testONNXModels("less");
 }
 
+TEST_P(Test_ONNX_layers, Compare_GTorEQ)
+{
+    testONNXModels("greater_or_equal");
```
Comment (rogday) [inline modules/dnn/test/test_onnx_importer.cpp:687]: Looking at `Compare_LT` one might assume that the same skips should be performed for the inference engine. Did you test it?

**Review by rogday — COMMENTED** [review]
Looks good, one question though.

File: modules/dnn/test/test_onnx_importer.cpp (hunk 1 above)
Comment (alalek) [reply modules/dnn/test/test_onnx_importer.cpp:687]: It could be ignored as we don't have regular stable CI with this configuration.

Comment (zihaomu) [reply modules/dnn/test/test_onnx_importer.cpp:687]: Currently, all compare operators have been mapped to `nary_eltwise_layers.cpp`. And `nary_eltwise_layers.cpp` has no inference engine supported.

**Review by fengyuentau — COMMENTED** [review]
Looks good. But how do we deal with the output type? In the model from https://github.com/opencv/opencv_zoo/pull/82, the GreaterOrEqual operator is followed by a Cast operator, which casts the boolean output from GreaterOrEqual to float. I dont think this casting problem can be solved with this pull request, isnt it?

**Comment by zihaomu** [comment]
> the `GreaterOrEqual` operator is followed by a Cast operator, which casts the boolean output from GreaterOrEqual to float. I dont think this casting problem can be solved with this pull request, isnt it?

For now, the `GreaterOrEqual` output the float32 instead of boolean format. So the `cast` do nothing. I am not sure if I am correct. More details can be found at [this part](https://github.com/opencv/opencv/blob/4.x/modules/dnn/src/layers/nary_eltwise_layers.cpp#L204-L256).

**Comment by fengyuentau** [comment]
> So the cast do nothing

I just checked `parseCast`. It should do nothing indeed (see line 2608).
https://github.com/opencv/opencv/blob/347246901eccabe503985a64f16813ca859af25a/modules/dnn/src/onnx/onnx_importer.cpp#L2579-L2610

**Review by rogday — APPROVED** [review]
:+1:

**Review by fengyuentau — APPROVED** [review]
👍👍

**Pull request merged**